### DISCUSSION TITLE
manually override image size

### DISCUSSION THREAD

**Discussion by frobert** [codingjoe/django-pictures#147] [Q&A] [answered]
Hi, is it possible to manually override image width and height in template?

Thanks.

**Comment by codingjoe** [comment]
Hi @frobert,

As mentioned in my previous comment, it takes a little getting used to. Before hard-coding sizes, I recommend reading a little into HTMLs picture tag: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture

However, there are some legacy use cases and we have a solution for those. You can read up more on this here: https://github.com/codingjoe/django-pictures?tab=readme-ov-file#legacy-use-cases-email

Cheers!
Joe

**Comment by atnartur** [comment]
Hi @codingjoe,

The library offers extensive functionality and simplifies the transition from legacy `<img>` tags to the modern `<picture>` tags. While I appreciate the push towards using `<picture>`, there are numerous scenarios where explicitly setting image sizes is beneficial.

Currently, the `<picture>` tag inherits the original image size from the `<img>` tag, affecting the resulting container as well. So every image insertion will have the original image's size on website in some cases. Although reverting to the legacy implementation is an option, it undermines the benefits of upgrading from django-stdimage to django-pictures.

What do you think about adding an ability to set explicit image size from template tag?

**Reply by codingjoe** [reply]
Hi @atnartur,

Thanks for sharing your experiences, and I appreciate your kind words. Just to clarify, you are talking about `picture img[width]` or `picture img[height]` when you are talking about image size, right?

I checked the code, this would in fact currently not possible. It would be a reasonably simple change, though. If this is, indeed, what you need, I can help you create a patch. Of course, I can also write a patch myself, but I would rather not steal your thunder.

Cheers!
Joe

**Reply by atnartur** [reply]
Using the modern approach with aspect ratio and some size of image, I want to set only img[width] or img[height] in a specific place. This value alongside the ratio will give all benefits of using `<picture>` tag. If you agree, I can make a pull request with a change, and we will discuss further details there.

**Reply by codingjoe** [reply, marked as answer]
Sure, I created a ticket #176
